include/forcing/NetCDFPerFeatureDataProvider.hpp

```diff
@@ -104,69 +162,81 @@ namespace data_access
+            // if absent, assume seconds
+            double time_scale_factor = 1;
+            time_unit = TIME_SECONDS;
+            try {
+                auto time_unit_att = time_var.getAtt("units");
 
-            // if time att is not encoded 
-            // TODO determine how this should be handled
-            std::string time_unit_str;
-            double time_scale_factor;     
+                // if time att is not encoded 
+                // TODO determine how this should be handled
+                std::string time_unit_str;
 
-            if ( time_unit_att.isNull() )
-            {
-                time_unit = TIME_SECONDS;
-                time_scale_factor = 1;
-            }
-            else
-            {  
-                time_unit_att.getValues(time_unit_str);
-            }
+                if ( time_unit_att.isNull() )
+                {
+                    log_stream << "Warning using defualt time units\n";
+                }
+                else
+                {  
+                    time_unit_att.getValues(time_unit_str);
+                }
 
-            // set time unit and scale factor
-            if ( time_unit_str == "h" || time_unit_str == "hours ")
-            {
-                time_unit = TIME_HOURS;
-                time_scale_factor = 3600;
-            }
-            else if ( time_unit_str == "m" || time_unit_str == "minutes" )
-            {
+                // set time unit and scale factor
+                if ( time_unit_str == "h" || time_unit_str == "hours ")
+                {
+                    time_unit = TIME_HOURS;
+                    time_scale_factor = 3600;
+                }
+                else if ( time_unit_str == "m" || time_unit_str == "minutes" )
+                {
                     time_unit = TIME_MINUETS;
                     time_scale_factor = 60;
-            }
-            else if ( time_unit_str ==  "s" || time_unit_str == "seconds" )
-            {
+                }
+                else if ( time_unit_str ==  "s" || time_unit_str == "seconds" )
+                {
                     time_unit = TIME_SECONDS;
                     time_scale_factor = 1;
-            }
-            else if ( time_unit_str ==  "ms" || time_unit_str == "milliseconds" )
-            {
+                }
+                else if ( time_unit_str ==  "ms" || time_unit_str == "milliseconds" )
+                {
                     time_unit = TIME_MILLISECONDS;
                     time_scale_factor = .001;
-            }
-            else if ( time_unit_str ==  "us" || time_unit_str == "microseconds" )
-            {
+                }
+                else if ( time_unit_str ==  "us" || time_unit_str == "microseconds" )
+                {
                     time_unit = TIME_MICROSECONDS;
                     time_scale_factor = .000001;
+                }
+                else if ( time_unit_str ==  "ns" || time_unit_str == "nanoseconds" )
+                {
```

Comment:
This doesn't seem appropriate.  If the time units are reported as `nanoseconds`, then why use `seconds`?